### PR TITLE
Update AnalyzerRunner to include a netcoreapp3.0 target

### PR DESCRIPTION
**Pull request by sharwell**
This is now blocked on:

#40885 
#40886 
#40888

### CODE REVIEW

File: src/Tools/AnalyzerRunner/Program.cs [hunk 1]
```diff
@@ -45,7 +45,21 @@ public static async Task Main(string[] args)
                     cts.Cancel();
                 };
 
+#if NETCOREAPP
+            // Since we are running as a dotnet tool we should be able to find an instance of
+            // MSBuild in a .NET Core SDK.
+            var msBuildInstance = MSBuildLocator.QueryVisualStudioInstances().First();
```
Comment (jasonmalinowski) [inline src/Tools/AnalyzerRunner/Program.cs]: FYI this is conflicting with your other PR.

File: src/Tools/AnalyzerRunner/PerformanceTracker.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+using System;
+using System.Diagnostics;
+
+namespace AnalyzerRunner
+{
+    internal sealed class PerformanceTracker
+    {
+        private readonly Stopwatch _stopwatch;
+        private readonly long _initialAllocation;
```
Comment (jasonmalinowski) [inline src/Tools/AnalyzerRunner/PerformanceTracker.cs]:
```suggestion
        private readonly long _initialTotalAllocatedBytes;
```
?

File: src/Tools/AnalyzerRunner/Program.cs [hunk 3]
```diff
@@ -65,9 +79,15 @@ public static async Task Main(string[] args)
...
 
-            Stopwatch stopwatch = Stopwatch.StartNew();
+            var stopwatch = PerformanceTracker.StartNew();
             var properties = new Dictionary<string, string>
             {
+#if NETCOREAPP
+                // This property ensures that XAML files will be compiled in the current AppDomain
+                // rather than a separate one. Any tasks isolated in AppDomains or tasks that create
+                // AppDomains will likely not work due to https://github.com/Microsoft/MSBuildLocator/issues/16.
+                { "AlwaysCompileMarkupFilesInSeparateDomain", bool.FalseString },
```
Comment (jasonmalinowski) [inline src/Tools/AnalyzerRunner/Program.cs]: Should this be moved to MSBuildWorkspace?

File: src/Tools/AnalyzerRunner/LooseAssemblyVersionLoader.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+                    return LoadAndCache(context, candidatePath);
+                }
+            }
+
+            return null;
+        }
+
+        /// <remarks>
+        /// Assumes we have a lock on _guard.
```
Comment (jasonmalinowski) [inline src/Tools/AnalyzerRunner/LooseAssemblyVersionLoader.cs]: Our convention in other parts of the codebase is to suffix the method with _NoLock which indicates "this doesn't take a lock, you had better have done it already".

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj [hunk 5]
```diff
@@ -4,23 +4,21 @@
...
-
   <ItemGroup>
     <PackageReference Include="SQLitePCLRaw.bundle_green" Version="$(SQLitePCLRawbundle_greenVersion)" PrivateAssets="all" />
     <PackageReference Include="Microsoft.VisualStudio.Composition" Version="$(MicrosoftVisualStudioCompositionVersion)" />
     <PackageReference Include="Microsoft.Build.Locator" Version="$(MicrosoftBuildLocatorVersion)" />
     <PackageReference Include="System.Buffers" Version="$(SystemBuffersVersion)" />
-    <PackageReference Include="SourceBrowser" Version="$(SourceBrowserVersion)" ExcludeAssets="build;compile" PrivateAssets="all" />
     <PackageReference Include="System.Buffers" Version="$(SystemBuffersVersion)" />
+    <PackageReference Include="System.ComponentModel.Composition" Version="$(SystemComponentModelCompositionVersion)" />
+    <PackageReference Include="Microsoft.Win32.Registry" Version="$(MicrosoftWin32RegistryVersion)" />
```
Comment (jasonmalinowski) [inline src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: What needed this? Or is this leftover from something earlier?

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj [hunk 6]
```diff
@@ -34,15 +32,22 @@
...
     <ProjectReference Include="..\..\EditorFeatures\CSharp\Microsoft.CodeAnalysis.CSharp.EditorFeatures.csproj" />
     <ProjectReference Include="..\..\EditorFeatures\VisualBasic\Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.vbproj" />
-    <ProjectReference Include="..\..\Features\CSharp\Portable\Microsoft.CodeAnalysis.CSharp.Features.csproj" />
-    <ProjectReference Include="..\..\Features\VisualBasic\Portable\Microsoft.CodeAnalysis.VisualBasic.Features.vbproj" />
   </ItemGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net472'">
+    <!-- Install the AppDomainManager from SourceBrowser https://github.com/microsoft/MSBuildLocator/issues/16 -->
+    <PackageReference Include="SourceBrowser" Version="$(SourceBrowserVersion)" ExcludeAssets="build;compile" PrivateAssets="all" />
```
Comment (jasonmalinowski) [inline src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: You said there's a bug tracking moving whatever workaround this is into MSBuildWorkspace proper?

File: src/Tools/AnalyzerRunner/Program.cs (hunk 1 above)
Comment (sharwell) [reply src/Tools/AnalyzerRunner/Program.cs]: Hence why I've marked this pull request as blocked 😄

File: src/Tools/AnalyzerRunner/Program.cs (hunk 3 above)
Comment (sharwell) [reply src/Tools/AnalyzerRunner/Program.cs]: I'm not ready to make that proposal yet.

File: src/Tools/AnalyzerRunner/LooseAssemblyVersionLoader.cs (hunk 4 above)
Comment (sharwell) [reply src/Tools/AnalyzerRunner/LooseAssemblyVersionLoader.cs]: This file is taken verbatim from dotnet/format.

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj (hunk 5 above)
Comment (sharwell) [reply src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: Resolves a build error related to assembly versions.

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj (hunk 6 above)
Comment (sharwell) [reply src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: :arrow_right: #36165

File: src/Tools/AnalyzerRunner/LooseAssemblyVersionLoader.cs (hunk 4 above)
Comment (JoeRobich) [reply src/Tools/AnalyzerRunner/LooseAssemblyVersionLoader.cs]: Didn't know about that convention. If I am ever back in dotnet/format, I'll work that in. =)

File: src/Tools/AnalyzerRunner/PerformanceTracker.cs (hunk 2 above)
Comment (sharwell) [reply src/Tools/AnalyzerRunner/PerformanceTracker.cs]: Updated

File: src/Tools/AnalyzerRunner/LooseAssemblyVersionLoader.cs (hunk 4 above)
Comment (sharwell) [reply src/Tools/AnalyzerRunner/LooseAssemblyVersionLoader.cs]: Updated naming here to follow Roslyn convention

**Pull request merged**